SDWebImage/SDAnimatedImageView.m

```diff
@@ -240,17 +243,38 @@ - (void)setImage:(UIImage *)image
 }
 
 #if SD_UIKIT
-- (void)setRunLoopMode:(NSString *)runLoopMode
+- (void)setRunLoopMode:(NSRunLoopMode)runLoopMode
 {
-    if (![@[NSDefaultRunLoopMode, NSRunLoopCommonModes] containsObject:runLoopMode]) {
```

Comment:
There logic, seems does not handle the case, when user change `runLoopMode` after animation started. Seems we can do better. Like YYImage.